GVFS/GVFS.Common/NuGetUpgrade/NuGetUpgrader.cs

```diff
@@ -605,6 +591,29 @@ private bool TryRefreshCredentials(out string error)
                 this.TraceException(ex, nameof(this.TryRefreshCredentials), "Failed to refresh credentials.");
                 return false;
             }
+        }
+
+        private bool EnsureNuGetFeedInitialized(out string error)
+        {
+            if (!this.isNuGetFeedInitialized)

```

Comment:
Should this not be an error case when initialising twice?

Reply:
We are making this call each time we attempt to query the remote endpoint, so it is just to ensure that we are initialized already. It is an internal method, and the parent method could potentially be called multiple times...